storage/rocksdb/clone/client.cc

```diff
@@ -809,6 +818,9 @@ int rocksdb_clone_apply(handlerton *, THD *thd, const uchar *loc, uint loc_len,
       return client::instance().save_and_return_error(
           ER_CLONE_PROTOCOL, "Received invalid clone data descriptor");
   }
+
+  assert(false);
```

Comment:
include/my_compiler.h has MY_ASSERT_UNREACHABLE(). Should we use that instead?

Reply:
Good catch